### DISCUSSION TITLE
Renovate cannot resolve GitHub workflows located in a different GitHub organization.

### DISCUSSION THREAD

**Discussion by srmadsen** [renovatebot/renovate#19760] [Request Help]
We have multiple GitHub organizations in our GitHub Enterprise Cloud.  For security reasons, we place all our internal GitHub Workflows in one organization (fs-cicd), and all other organizations (fs-eng, fs-webdev, etc.) reference them from that repository.

So, when running a workflow in an fs-eng organization repository that references a workflow in the fs-cicd organization, we get the following error in the renovate logs:

`Datasource unknown error
{
  "datasource": "github-tags",
  "packageName": "fs-cicd/paas-cicd-workflows",
  "err": {
    "message": "Could not resolve to a Repository with the name 'fs-cicd/paas-cicd-workflows'.",
    "stack": "Error: Could not resolve to a Repository with the name 'fs-cicd/paas-cicd-workflows'.\n    at GithubGraphqlDatasourceFetcher.doRawQuery (/home/ubuntu/renovateapp/node_modules/renovate/dist/util/github/graphql/datasource-fetcher.js:79:29)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async GithubGraphqlDatasourceFetcher.doShrinkableQuery (/home/ubuntu/renovateapp/node_modules/renovate/dist/util/github/graphql/datasource-fetcher.js:130:26)\n    at async GithubGraphqlDatasourceFetcher.doPaginatedQuery (/home/ubuntu/renovateapp/node_modules/renovate/dist/util/github/graphql/datasource-fetcher.js:161:33)\n    at async GithubGraphqlDatasourceFetcher.getItems (/home/ubuntu/renovateapp/node_modules/renovate/dist/util/github/graphql/datasource-fetcher.js:197:21)\n    at async Function.query (/home/ubuntu/renovateapp/node_modules/renovate/dist/util/github/graphql/datasource-fetcher.js:29:23)\n    at async queryTags (/home/ubuntu/renovateapp/node_modules/renovate/dist/util/github/graphql/index.js:8:17)\n    at async GithubTagsDatasource.getReleases (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/github-tags/index.js:58:28)\n    at async getRegistryReleases (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/index.js:57:17)\n    at async fetchReleases (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/index.js:241:23)\n    at async getPkgReleases (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/index.js:295:34)\n    at async lookupUpdates (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/process/lookup/index.js:53:45)\n    at async fetchDepUpdates (/home/ubuntu/renovateapp/node_modules/renovate/dist/workers/repository/process/fetch.js:52:25)\n    at async /home/ubuntu/renovateapp/node_modules/p-map/index.js:57:22"
  }
}`

My current best guess is that Renovate does not support workflow references to workflows in a different repository.  If I'm wrong, please correct me.

**Comment by rarkins** [comment]
Perhaps it's trying to hit GitHub.com

**Reply by srmadsen** [reply]
@rarkins -- We use GitHub Enterprise Cloud to host all our organizations source.  So, yes, it is accessing GitHub.com.  Renovate is being used with success in all orgs.  However, it does not appear to support cross-org access to github workflows -- which github workflows supports.

I would love to see the GraphQL query behind this, but I'm betting that it does not expect to see an "org/repo" as a Repository name.

**Reply by srmadsen** [reply]
@rarkins -- Or perhaps Renovate only has access to the code for the repository it is processing and cannot find references outside of that context?

**Reply by rarkins** [reply]
Are you self-hosting, or using the Mend Renovate app?

**Reply by srmadsen** [reply]
We use the Mend Renovate App

**Reply by rarkins** [reply]
Is the other workflow you're trying to resolve private?

**Reply by rarkins** [reply]
And assuming they're private, what you're seeing makes complete sense. Of course GitHub doesn't let a token for the `fs-eng` org access private repos in other orgs such as `fs-cicd`. You would need to create something like a PAT with read-only access to the necessary repos and then configure it using hostRules

**Reply by srmadsen** [reply]
That explains what is happening.  Thanks.

Next question.  When will Renovate support the GitHub Enterprise?

**Reply by rarkins** [reply]
Renovate has supported GitHub Enterprise since 2018. But if you plan to be more specific on what you mean, or what you're missing, please start a new discussion.

**Reply by srmadsen** [reply]
Hmmm.  Currently we install a Renovate GitHub App in every GitHub organization.  That leads to this problem where the Renovate GitHub app cannot access repositories in different organization.  Instead, what we really need to do is install ONE GitHub App in the enterprise so that the Renovate GitHub app can access all organizations in the enterprise.  Is that supported?

**Reply by srmadsen** [reply]
Oops.  I missed your request to move enterprise discussion to a separate issue.

**Reply by rarkins** [reply]
It's ok as the question is quite closely related. What you're asking for is not supported by GitHub so there's no possibility for Renovate to support that. Instead you'd need to run with a PAT which has server wide access.
